src/path.cc

```diff
@@ -268,4 +267,44 @@ std::string PathResolve(Environment* env,
 }
 #endif  // _WIN32
 
+void ToNamespacedPath(Environment* env, BufferValue* path) {
+#ifdef _WIN32
+  if (path->length() == 0) return;
+  auto resolved_path = node::PathResolve(env, {path->ToStringView()});
```

Comment:
@anonrig You are assuming that node::PathResolve is equivalent to JavaScript's win32.resolve. I suggest that this should be tested explicitly if it isn't.

Importantly, the JavaScript version calls explicitly on the current working directory:

```JavaScript
path = process.env[`=${resolvedDevice}`] || process.cwd();
```

The C++ code goes for...

```C++
auto cwd = env->GetCwd(env->exec_path());
```

This may or many not be equivalent. :-/ It would be nice to have explicit checks.

Reply:
(Note that I do not expect the code to be wrong.)